### ISSUE TITLE
When a table's caption is positioned absolutely, NVDA incorrectly counts an extra row

### ISSUE THREAD

**Issue by nvaccessAuto** [nvaccess/nvda#3813]
**Reported by hhillen on 2014-01-24 11:46**
When a table has a caption element that is styled with position:absolute, NVDA will treat it as a row when calculating the total number of rows in the table. A common use case for using absolute positioning is when the caption should be hidden visually but not from assistive technology.

I have attached a [test case](https://www.nvaccess.org/files/nvdaTracAttachments/3813/testcase_absolutly_positioned_caption.html) that illustrates this. The page contains two tables with identical content, except that in the second table the caption text is hidden off screen using CSS. When pressing the T key twice, NVDA will correctly announce 4 rows for the first table and incorrectly 5 rows for the second table.

**Comment by vimalans** [comment]
Is there any update on this issue? I encountered this problem today with FF and NVDA.

**Comment by jcsteh** [comment]
As far as I know, a Mozilla bug was never filed for this. Feel free to do so and cc me as explained above.

**Comment by luisluix** [comment]
its may 2018, are there any updates for this?

**Comment by jcsteh** [comment]
Actually, there is a Mozilla bug for this: [bug 995888](https://bugzilla.mozilla.org/show_bug.cgi?id=995888).

**Comment by Adriani90** [comment]
I have edited the description and updated the link so it works again. The issue is still reproducible in NVDA 2019.1.1 only on Firefox. Testing in Firefox 66.0.5. Chrome and IE work as expected.

**Comment by Adriani90** [comment]
I cannot reproduce this in Firefox 110.0.1 with NVDA last alpha. @jcsteh maybe it is worthy for bugzilla. I have accessibility cache enabled.

**Comment by jcsteh** [comment]
It is fixed with the cache enabled. However, it is not fixed (and will not be fixed) with the cache disabled. The cache is still not enabled by default on release, though it hopefully will be in the next month or two.

**Comment by jcsteh** [comment]
Now that the cache is enabled for all users, this is fixed in Firefox.
